internal/ingest/linksystem.go

```diff
@@ -242,29 +241,23 @@ func (ing *Ingester) ingestAd(publisherID peer.ID, adCid cid.Cid, ad schema.Adve
 func (ing *Ingester) ingestEntryChunk(ctx context.Context, publisher peer.ID, adCid cid.Cid, ad schema.Advertisement, entryChunkCid cid.Cid) error {
 	log := log.With("publisher", publisher, "adCid", adCid, "cid", entryChunkCid)
 
-	// Get data corresponding to the block.
-	entryChunkKey := dsKey(entryChunkCid.String())
-	val, err := ing.ds.Get(ctx, entryChunkKey)
+	chunk, err := ing.loadEntryChunk(entryChunkCid)
 	if err != nil {
-		return fmt.Errorf("cannot fetch the node from datastore: %w", err)
+		return fmt.Errorf("failed to load chunk from datastore ipldNode: %w", err)
 	}
+
 	defer func() {
 		// Remove the content block from the data store now that processing it
 		// has finished. This prevents storing redundant information in several
 		// datastores.
+		entryChunkKey := dsKey(entryChunkCid.String())
```

Comment:
Since `chunkKey` has to be computed twice, here and in `loadEntryChunk`, it would be better to compute it once and pass it into `loadEntryChunk`.  That will avoid an additional `dsKey.Clean`